core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.kt

```diff
@@ -83,20 +84,24 @@ open class KiwixWebView @SuppressLint("SetJavaScriptEnabled") constructor(
     clearCache(true)
     setWebViewClient(webViewClient)
     webChromeClient = KiwixWebChromeClient(callback, nonVideoView, videoView, this).apply {
-      setOnToggledFullscreen { fullscreen ->
-        setWindowVisibility(
-          if (fullscreen) View.SYSTEM_UI_FLAG_LOW_PROFILE else View.SYSTEM_UI_FLAG_VISIBLE
-        )
-        callback.onFullscreenVideoToggled(fullscreen)
-      }
+      setOnToggledFullscreen(object : ToggledFullscreenCallback {
+        override fun toggledFullscreen(fullscreen: Boolean) {
+          setWindowVisibility(
+            if (fullscreen) View.SYSTEM_UI_FLAG_LOW_PROFILE else View.SYSTEM_UI_FLAG_VISIBLE
+          )
+          callback.onFullscreenVideoToggled(fullscreen)
+        }
+      })
     }
   }
 
   override fun performLongClick(): Boolean {
     val result = hitTestResult
     if (result.type == HitTestResult.SRC_ANCHOR_TYPE) {
-      result.extra?.let(callback::webViewLongClick)
-      return true
+      result.extra?.let {
+        callback::webViewLongClick
+        return@performLongClick true
+      }
```

Comment:
Why are we getting true if it clicks only the `src` anchor type? That's not what this function is for.